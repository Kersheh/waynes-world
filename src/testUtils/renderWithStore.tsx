import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose, Reducer } from 'redux';
import createSagaMiddleware from 'redux-saga';
import { render } from '@testing-library/react';

import rootReducer from 'store/reducers';
import rootSaga from 'store/sagas';

const renderWithStore = (component: React.ReactElement) => {
  const sagaMiddleware = createSagaMiddleware();
  const store = createStore(
    rootReducer as Reducer,
    compose(applyMiddleware(sagaMiddleware))
  );
  sagaMiddleware.run(rootSaga);

  return render(<Provider store={store}>{component}</Provider>);
};

export default renderWithStore;
