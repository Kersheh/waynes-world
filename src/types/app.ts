export type View = 'home' | 'search' | 'library' | 'settings';
