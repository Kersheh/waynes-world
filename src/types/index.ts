import { RootState as RootReducerState } from 'store/reducers';

export * from './app';
export * from './library';
export * from './spotify';

export type RootState = RootReducerState;

export type Action = {
  type: string;
  data?: any;
};

export type RequestData = {
  [key: string]: any;
};
