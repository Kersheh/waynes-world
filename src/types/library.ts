export type Album = {
  id: string;
  album: string;
  artist: string;
  year?: number;
  genre?: string;
  shelf?: string;
  comments?: string;
  artBase64?: string;
  createdAt: string;
  updatedAt: string;
  favourite?: boolean;
};

export type EditAlbumFormValues = {
  artist: string;
  album: string;
  year: string;
  genre: string;
  shelf: string;
  comments: string;
};
