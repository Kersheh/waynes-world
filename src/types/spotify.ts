export type SpotifyImage = {
  height: number;
  width: number;
  url: string;
};

export type SpotifyArtist = {
  id: string;
  genres: Array<string>;
  images: Array<SpotifyImage>;
  name: string;
  type: string;
};

export type SpotifyAlbum = {
  id: string;
  album_type: string;
  artists: Array<{
    id: string;
    name: string;
    type: string;
  }>;
  images: Array<SpotifyImage>;
  name: string;
  release_date: string;
  type: string;
};
