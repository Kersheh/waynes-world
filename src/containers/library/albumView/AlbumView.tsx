import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { parseISO, format } from 'date-fns';

import { Album } from 'types';
import Button from 'components/button/Button';
import Image from 'components/image/Image';
import IconCaret from 'components/icons/IconCaret';
import TextArea from 'components/textArea/TextArea';
import { clearAlbumView } from '../libraryActions';
import styles from './AlbumView.module.scss';

interface AlbumViewProps {
  album: Album;
}
const AlbumView = ({ album }: AlbumViewProps) => {
  const dispatch = useDispatch();
  const [backgroundColor, setBackgroundColor] = useState<string | null>(null);

  return (
    <div className={styles.albumView}>
      <div className={styles.header}>
        <Button onClick={() => dispatch(clearAlbumView())}>
          <IconCaret className={styles.left} />
        </Button>
      </div>

      <div
        className={styles.artWrapper}
        style={{
          backgroundImage: backgroundColor
            ? `linear-gradient(${backgroundColor}, black)`
            : ''
        }}
      >
        <div className={styles.art}>
          {!album.artBase64 && (
            <div className={styles.artMissing} draggable="false">
              Testing
            </div>
          )}
          {album.artBase64 && (
            <Image
              className={styles.artImg}
              src={`data:image/jpeg;base64,${album.artBase64}`}
              getImgDominantHexColor={setBackgroundColor}
            />
          )}
        </div>
      </div>

      <div className={styles.info}>
        <div className={styles.main}>
          <div className={styles.left}>
            <div className={styles.album}>{album.album}</div>
            <div className={styles.artist}>{album.artist}</div>
            {album.year && (
              <div className={styles.year}>Album Â· {album.year}</div>
            )}
            {album.genre && <div className={styles.genre}>{album.genre}</div>}
          </div>

          <div className={styles.right}>
            {album.createdAt && (
              <div className={styles.createdAt}>
                <div className={styles.label}>Added:</div>
                <div>{format(parseISO(album.createdAt), 'MMM dd, yyyy')}</div>
              </div>
            )}
            {album.shelf && (
              <div className={styles.shelf}>
                <div className={styles.label}>Shelf:</div>
                <div>{album.shelf}</div>
              </div>
            )}
          </div>
        </div>

        {album.comments && (
          <div className={styles.comments}>
            <hr />
            <TextArea name="comments" readOnlyValue={album.comments} />
          </div>
        )}
      </div>
    </div>
  );
};

export default AlbumView;
