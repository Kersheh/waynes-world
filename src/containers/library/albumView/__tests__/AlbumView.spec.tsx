import React from 'react';

import { Album } from 'types';
import renderWithStore from 'testUtils/renderWithStore';
import AlbumView from '../AlbumView';

describe('<AlbumView/> component', () => {
  it('should render', () => {
    const { container } = renderWithStore(<AlbumView album={{} as Album} />);

    expect(container).toMatchSnapshot();
  });
});
