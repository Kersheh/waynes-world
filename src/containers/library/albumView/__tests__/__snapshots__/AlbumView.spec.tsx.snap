// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlbumView/> component should render 1`] = `
<div>
  <div
    class="albumView"
  >
    <div
      class="header"
    >
      <button
        class="button text"
        type="button"
      >
        <svg
          aria-labelledby="fqsi-ant-right-title"
          class="left"
          id="si-ant-right"
          viewBox="0 0 574 1024"
        >
          <path
            d="M10 9Q0 19 0 32t10 23l482 457L10 969Q0 979 0 992t10 23q10 9 24 9t24-9l506-480q10-10 10-23t-10-23L58 9Q48 0 34 0T10 9z"
          />
        </svg>
      </button>
      <div
        class="right"
      >
        <button
          class="button icon"
          type="button"
        >
          <svg
            fill="white"
            height="18px"
            viewBox="0 0 24 24"
            width="18px"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16.5 3c-1.74 0-3.41.81-4.5 2.09C10.91 3.81 9.24 3 7.5 3 4.42 3 2 5.42 2 8.5c0 3.78 3.4 6.86 8.55 11.54L12 21.35l1.45-1.32C18.6 15.36 22 12.28 22 8.5 22 5.42 19.58 3 16.5 3zm-4.4 15.55l-.1.1-.1-.1C7.14 14.24 4 11.39 4 8.5 4 6.5 5.5 5 7.5 5c1.54 0 3.04.99 3.57 2.36h1.87C13.46 5.99 14.96 5 16.5 5c2 0 3.5 1.5 3.5 3.5 0 2.89-3.14 5.74-7.9 10.05z"
            />
          </svg>
        </button>
        <button
          class="button icon"
          type="button"
        >
          <svg
            height="24"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zm2.46-7.12l1.41-1.41L12 12.59l2.12-2.12 1.41 1.41L13.41 14l2.12 2.12-1.41 1.41L12 15.41l-2.12 2.12-1.41-1.41L10.59 14l-2.13-2.12zM15.5 4l-1-1h-5l-1 1H5v2h14V4z"
              fill="white"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="artWrapper"
    >
      <div
        class="art"
      >
        <div
          class="artMissing"
          draggable="false"
        >
          Testing
        </div>
      </div>
    </div>
    <div
      class="info"
    >
      <div
        class="main"
      >
        <div
          class="left"
        >
          <div
            class="album"
          />
          <div
            class="artist"
          />
        </div>
        <div
          class="right"
        />
      </div>
    </div>
  </div>
</div>
`;
